runtime/lua/vim/lsp/util.lua

```diff
@@ -147,6 +147,92 @@ local function sort_by_key(fn)
   end
 end
 
+---@private
+--- Gets the zero-indexed lines from the given buffer.
+--- Works on unloaded buffers by reading the file using libuv to bypass buf reading events.
+--- Falls back to loading the buffer and nvim_buf_get_lines for buffers with non-file URI.
+---
+---@param bufnr number bufnr to get the lines from
+---@param rows number[] zero-indexed line numbers
+---@return table<number string> a table mapping rows to lines
+local function get_lines(bufnr, rows)
```

Comment:
Changed this from `uri` to `bufnr`. Otherwise it wouldn't work for unsaved buffers (which we use a lot in tests)
Also made it private (which is we it moved up, because it is used below and there isn't a M. forward reference anymore)